src/base/settingsstorage.cpp

```diff
@@ -26,7 +26,9 @@
  * but you are not obligated to do so. If you do not wish to do so, delete this
  * exception statement from your version.
  */
+#include "settingsstorage.h"
```

Comment:
Your example does not prove the efficiency of this.
>This mistake is easily spotted in case of the proper implementation file.

This mistake is easily spotted in any case (since we include file.h in some other files).